org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/commands/ProjectCommand.java

```diff
@@ -128,6 +130,17 @@ public static boolean isTestFile(String uri) throws CoreException {
 		return false;
 	}
 
+	public static List<URI> getAllJavaProjects() {
+		List<URI> javaProjects = new LinkedList<>();
+		for (IJavaProject javaProejct : ProjectUtils.getJavaProjects()) {
+			if (ResourcesPlugin.getWorkspace().getRoot().getProject(ProjectsManager.DEFAULT_PROJECT_NAME).equals(javaProejct.getProject())) {
```

Comment:
Why don't you return the default project? Might be valuable information to get for standalone Java files

Reply:
Because I'm thinking that the project path of the default project is totally different from that of the standalone Java files. The client side needs to have a way to resolve such mapping...

Reply:
well, no mapping = default project, no?

Reply:
Yes, I also thinks of that actually. What I'm afraid is, it might be in some situation that the client side cache is out of sync?

Hmm, Can I say that, if a java file does not belongs to any workspace folder, then it belongs to the default project?

Reply:
I think that's probably an accurate assumption.